src/main/java/com/ecore/roles/web/MembershipsApi.java

```diff
@@ -0,0 +1,17 @@
+package com.ecore.roles.web;
+
+import com.ecore.roles.web.dto.MembershipDto;
+import org.springframework.http.ResponseEntity;
+
+import java.util.List;
+import java.util.UUID;
+
+public interface MembershipsApi {
+
+    ResponseEntity<MembershipDto> assignRoleToMembership(
```

Comment:
It could be just called "assignRoleAndCreate" because it's inside of the membership api.